.github/workflows/python-wheels-publish-test.yml

```
@@ -0,0 +1,95 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) Contributors to the OpenEXR Project.
+
+name: Publish python distribution 📦 to TestPyPI
+
+on:
+
+  # Publish python wheels to test.pypi when a release candidate is tagged, 
+  # e.g. v3.4.5-rc, v3.4.5-rc6, etc.
+
+  push:
+    tags:
+      - v3.[0-9]+.[0-9]+-rc*
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    name: Python Wheels - ${{ matrix.os }} 
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+
+    environment:
+      name: testpypi
+      url: https://test.pypi.org/p/openexr
+
+    permissions:
+      id-token: write
+      
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Python 
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
```

```suggestion
          python-version: '3.x'

      - name: Create sdist
        # Only create it once.
        if: ${{ matrix.os == 'ubuntu-latest' }}
        run: pipx run build --sdist . --outdir wheelhouse
```
Generating and publishing an sdist on PyPI allows users to still be able to `pip install` even if no wheels match their environment. it also allows users that don't trust pre-compiled binaries on PyPI to just `pip install openexr --no-binary=:all:` which will instruct pip to only download sdists and compile everything locally.

(The same comment also applies to the two other workflows).

Note, I should note that as things are, CMake will download and build the latest Imath, which kind of makes the sdist completely non-reproducible.

---

Only the build on the OpenEXR main branch pulls from the Imath main branch, release branches of OpenEXR pull Imath from a specific tag, so a build via sdisted source should at least do that.

---

Umm, ok. How does that work?

---

The release branch just hard-codes a fixed tag: https://github.com/AcademySoftwareFoundation/openexr/blob/7adf2c8f6f11c650c304e70d286b49c1d5d55abb/cmake/OpenEXRSetup.cmake#L236